src/HexagonOffload.cpp

```diff
@@ -899,9 +909,13 @@ Buffer<uint8_t> compile_module_to_hexagon_shared_object(const Module &device_cod
     }
 
     // Link into a shared object.
+    std::string soname = "lib" + device_code.name() + ".so";
     Elf::HexagonLinker linker(device_code.target());
-    std::vector<std::string> dependencies = { "libhalide_hexagon_remote_skel.so" };
-    std::vector<char> shared_object = obj->write_shared_object(&linker, dependencies);
+    std::vector<std::string> dependencies = {
+        "lib" + runtime_module_name + ".so",
```

Comment:
This is a little dumb but doesn't seem to hurt anything, it *does* technically depend on itself :)